src/main/java/org/jabref/gui/entryeditor/LatexCitationsTabViewModel.java

```diff
@@ -164,15 +182,23 @@ public void setLatexDirectory() {
         DirectoryDialogConfiguration directoryDialogConfiguration = new DirectoryDialogConfiguration.Builder()
                 .withInitialDirectory(directory.get()).build();
 
-        dialogService.showDirectorySelectionDialog(directoryDialogConfiguration).ifPresent(selectedDirectory ->
-                databaseContext.getMetaData().setLatexFileDirectory(preferencesService.getFilePreferences().getUserAndHost(), selectedDirectory.toAbsolutePath()));
+        fileUpdateMonitor.removeListener(directory.get(), this::refreshLatexDirectory);
+
+        dialogService.showDirectorySelectionDialog(directoryDialogConfiguration).ifPresent(selectedDirectory -> {
+                    databaseContext.getMetaData().setLatexFileDirectory(preferencesService.getFilePreferences().getUserAndHost(), selectedDirectory.toAbsolutePath());
+                    try {
+                        fileUpdateMonitor.addListenerForFile(selectedDirectory, this::refreshLatexDirectory);
```

Comment:
I think, there should be a seperate method:

1. detach the old listener
2. attach a new listener
3. Call `refreshLatexDirectory`